frontend/src/components/homepage/homepage.tsx

```diff
@@ -13,24 +14,41 @@ const Homepage: React.FC = () => {
   const dispatch = useDispatch();
   const hasPodcasts = Boolean(podcasts.length);
 
+  const [filter, setFilter] = useState<Array<Podcast>>(podcasts);
+
   useEffect(() => {
     dispatch(homepageActions.loadPodcasts());
   }, []);
 
+  const handleChange = (searchValue: SearchPayload): void => {
+    const { search } = searchValue;
+    const filtered = filterPodcasts(search);
+    search === '' ? setFilter(podcasts) : setFilter(filtered);
```

Comment:
Don't use immediate invocation
```suggestion
    const payload = !search ? podcasts : filtered
    setFilter(payload)
```